pkg/scheduler/apis/config/types.go

```diff
@@ -270,3 +271,53 @@ func (p *Plugins) Append(src *Plugins) {
 	p.PostBind = appendPluginSet(p.PostBind, src.PostBind)
 	p.Unreserve = appendPluginSet(p.Unreserve, src.Unreserve)
 }
+
+// Apply merges custom plugins into default plugins.
+func (p *Plugins) Apply(customPlugins *Plugins) {
+	if customPlugins == nil {
+		return
+	}
+
+	p.QueueSort = mergePluginSets(p.QueueSort, customPlugins.QueueSort)
+	p.PreFilter = mergePluginSets(p.PreFilter, customPlugins.PreFilter)
+	p.Filter = mergePluginSets(p.Filter, customPlugins.Filter)
+	p.PostFilter = mergePluginSets(p.PostFilter, customPlugins.PostFilter)
+	p.Score = mergePluginSets(p.Score, customPlugins.Score)
+	p.Reserve = mergePluginSets(p.Reserve, customPlugins.Reserve)
+	p.Permit = mergePluginSets(p.Permit, customPlugins.Permit)
+	p.PreBind = mergePluginSets(p.PreBind, customPlugins.PreBind)
+	p.Bind = mergePluginSets(p.Bind, customPlugins.Bind)
+	p.PostBind = mergePluginSets(p.PostBind, customPlugins.PostBind)
+	p.Unreserve = mergePluginSets(p.Unreserve, customPlugins.Unreserve)
+}
+
+func mergePluginSets(defaultPluginSet, customPluginSet *PluginSet) *PluginSet {
+	if customPluginSet == nil {
+		customPluginSet = &PluginSet{}
+	}
+
+	if defaultPluginSet == nil {
+		defaultPluginSet = &PluginSet{}
+	}
+
+	disabledPlugins := sets.NewString()
+	for _, disabledPlugin := range customPluginSet.Disabled {
+		disabledPlugins.Insert(disabledPlugin.Name)
+	}
+
+	enabledPlugins := []Plugin{}
+	if !disabledPlugins.Has("*") {
+		for i := range defaultPluginSet.Enabled {
+			defaultEnabledPlugin := defaultPluginSet.Enabled[i]
+			if disabledPlugins.Has(defaultEnabledPlugin.Name) {
```

Comment:
nit: `if !disabled.Has(...) { enabled = append(..`.

`continue` would make more sense if we wanted to avoid nesting. But a single line of code is just fine.

Reply:
Prefer to return first in "wrong" path

Reply:
Again, that argument makes sense for bigger blocks of code. But not big deal.

Reply:
> Again, that argument makes sense for bigger blocks of code. But not big deal.

Thanks